go-crud/main.go

```diff
@@ -22,9 +26,64 @@ func main() {
 		return
 	}
 
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+
 	file := data.NewStore(*path)
+	file.ReadFromFile()
+
+	// ticker := time.NewTicker(5 * time.Second)
+
+	go func() {
```

Comment:
How about something like this?
```suggestion
	handler := books.Router(file)

	s := &http.Server{
		Addr:         ":8080",
		Handler:      handler,
		WriteTimeout: 2 * time.Second,
	}

	ticker := time.NewTicker(5 * time.Second)
	defer ticker.Stop()

	go func() {
		for {
			select {
			case <-ticker.C:
				file.WriteToFile()
				continue
			case <-sigs:
				file.WriteToFile()
				break
			}

		}

		// os.Exit(0)
		tc, cancel := context.WithTimeout(context.Background(), 5*time.Second)
		defer cancel()

		s.Shutdown(tc)
	}()
```